### DISCUSSION TITLE
Newbie: How to Match on Either type

### DISCUSSION THREAD

**Discussion by khtan** [vavr-io/vavr#2864] [Q&A] [answered]
In the documentation ( https://docs.vavr.io/#_patterns ), there is an example of matching the Option type :

Match(option).of(
    Case($Some($()), "defined"),
    Case($None(), "empty")
);

What is the canonical way to match Either<X,Y> type? I tried a similar pattern but it is not supported.

Either<String, Double> e = Right(12.0);

String x = Match(e).of(
    Case($Right($()), v -> v.get()),
    Case($Left($()), v -> v.getLeft().toString())
);

**Comment by danieldietrich** [comment, marked as answer]
Did you import the patterns from [io.vavr.Patterns](https://www.javadoc.io/static/io.vavr/vavr/0.10.3/io/vavr/Patterns.html)?
